dev-packages/cli/src/check-dependencies.ts

```diff
@@ -0,0 +1,306 @@
+        if (!packagesGroupedByName.has(name)) {
+            packagesGroupedByName.set(name, []);
+        }
+        const currentPackages = packagesGroupedByName.get(name)!;
+        currentPackages.push(currentPackage);
+        if (currentPackages.length > 1 && duplicates.indexOf(name) === -1) {
+            duplicates.push(name);
+        }
+    }
+
+    duplicates.sort();
+    const issues: DependencyIssue[] = [];
+    for (const duplicate of duplicates) {
+        const duplicatePackages = packagesGroupedByName.get(duplicate);
+        if (duplicatePackages && duplicatePackages.length > 0) {
+            issues.push({
+                issueType: 'multiple-versions',
+                package: duplicatePackages.pop()!,
+                relatedPackages: duplicatePackages,
+                severity: options.suppress ? 'warning' : 'error'
+            });
+        }
+    }
+
+    return issues;
+}
+
+function findTheiaVersionMix(packages: Package[], options: CheckDependenciesOptions): DependencyIssue[] {
+    // @theia/monaco-editor-core is following the versions of Monaco so it can't be part of this check
+    const theiaPackages = packages.filter(p => p.name.startsWith('@theia/') && !p.name.startsWith('@theia/monaco-editor-core'));
+    let theiaVersion = undefined;
+    let referenceTheiaPackage = undefined;
+    const packagesWithOtherVersion: Package[] = [];
+    for (const theiaPackage of theiaPackages) {
+        if (!theiaVersion && theiaPackage.version) {
+            theiaVersion = theiaPackage.version;
+            referenceTheiaPackage = theiaPackage;
+        } else if (theiaVersion !== theiaPackage.version) {
+            packagesWithOtherVersion.push(theiaPackage);
+        }
+    }
+
+    if (referenceTheiaPackage && packagesWithOtherVersion.length > 0) {
+        return [{
+            issueType: 'theia-version-mix',
+            package: referenceTheiaPackage,
+            relatedPackages: packagesWithOtherVersion,
+            severity: 'error'
+        }];
+    }
+    return [];
+}
+
+function printIssues(issues: DependencyIssue[]): void {
+    console.log();
+    const indent = issues.length.toString().length;
+    issues.forEach((issue, index) => {
+        printIssue(issue, index + 1, indent);
+    });
+}
+
+function printIssue(issue: DependencyIssue, issueNumber: number, indent: number): void {
+    const remainingIndent = indent - issueNumber.toString().length;
+    const indentString = ' '.repeat(remainingIndent + 1);
+    console.log(issueTitle(issue, issueNumber, indentString));
+    console.log(issueDetails(issue, '   ' + ' '.repeat(indent)));
+    console.log();
+}
+
+function issueTitle(issue: DependencyIssue, issueNumber: number, indent: string): string {
+    const dependent = issue.package.dependent ? ` in ${chalk.blueBright(issue.package.dependent ?? 'unknown')}` : '';
+    return chalk.bgWhiteBright.bold(`#${issueNumber}${indent}`) + ' ' + chalk.cyanBright(issue.package.name)
```

Comment:
I'm not sure if it is my setup or terminal preferences but I do not see the text (white on white) for the issue number:

![Screen Shot 2022-08-23 at 1 59 49 PM](https://user-images.githubusercontent.com/40359487/186231226-be904c56-e9bf-468a-aac2-e111cc12425f.png)

Reply:
Hm, interesting. I'll amend a change to set the font to black explicitly. Can you please test with your theme and terminal preferences, whether this works for you?